src/components/jobComponents/createJob.tsx

```diff
@@ -77,13 +80,19 @@ const CreateJob: React.FC = () => {
         description,
         userId: user.id,
         images: images.photos,
+        status,
       })
       .then(({ data }) => {
-        console.log(data);
         // get jobId for navigation
         const id = data.jobId;
-        // navigate to job details page
-        history.push(`/jobs/${id}`);
+        if (status === 'volunteer') {
+          // navigate to job details page
+          history.push(`/jobs/${id}`);
+        } else {
+          return new Promise(res => {
+            res(dispatch(fetchJob(id)));
+          }).then(() => history.push(`/checkout/${id}`));
```

Comment:
again - fetchJob already returns a promise

Reply:
you can not use a .then on dispatch(fetchJob(id))